src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -84,6 +85,13 @@ export class CalciteRadioButton {
   @Watch("name")
   nameChanged(newName: string): void {
     this.input.name = newName;
+    this.checkFirstRadioButton();
+    const currentValue: HTMLInputElement = document.querySelector(
+      `input[name="${this.name}"]:checked`
+    );
+    if (!currentValue?.value) {
+      this.uncheckAllRadioButtonsInGroup();
+    }
```

Comment:
@driskull this refactor ensures that when the `name` changes, not only will it ensure that the first `checked` radio in the group wins like it does on initial mount, but it will also guarantee that none of the calcite radios in a group remain in a "checked" visual state when their corresponding input isn't actually checked.  This was happening with some test code I was running to debug this issue:

```
setTimeout(() => {

    const tiles = document.querySelectorAll("input");
    if (tiles.length > 0) {
      setTimeout(() => {
        tiles.forEach((tile) => {
          tile.name = "foo";
          tile.checked = true;
        });
        console.log("names changed to foo")
      }, 3000);
      setTimeout(() => {
        tiles.forEach((tile) => {
          tile.checked = true;
          tile.name = "bar";
        });
        console.log("names changed to bar")
      }, 6000);

    }
  }, 1000)
```
☝️ These are contrived examples, but showcase that the radio will properly respond to both of these conditions.  This code also works, when the querySelector searches for `calcite-radio-button`s.  In the second setTimeout in the example above, when the name changes after the checked property, the entire radio group resets to have no checked value since the name changes last and the checked property is no longer valid since it was set before the name change and only applied to the old name.

Reply:
Sounds good!